diff --git a/ZeroBraneStudio-linux/app/zbstudio.sh b/ZeroBraneStudio-linux/app/zbstudio.sh
index c2af13f..d20bd2e 100755
--- a/ZeroBraneStudio-linux/app/zbstudio.sh
+++ b/ZeroBraneStudio-linux/app/zbstudio.sh
@@ -3,4 +3,4 @@
 if [[ "$(uname -m)" == "x86_64" ]]; then ARCH="x64"; else ARCH="x86"; fi
 CWD="$PWD" # save the current directory, as it's going to change
 
-(cd "/opt/zbstudio"; bin/linux/$ARCH/lua src/main.lua zbstudio -cwd "$CWD" "$@" 2>/dev/null) &
+(cd "/app/opt/zbstudio"; bin/linux/$ARCH/lua src/main.lua zbstudio -cwd "$CWD" "$@" 2>/dev/null) &
diff --git a/ZeroBraneStudio-linux/install.sh b/ZeroBraneStudio-linux/install.sh
index 32d13d4..b9fb89f 100755
--- a/ZeroBraneStudio-linux/install.sh
+++ b/ZeroBraneStudio-linux/install.sh
@@ -1,50 +1,40 @@
 #!/bin/bash
 
-INSTALL_PATH=/opt/zbstudio
+INSTALL_PATH=/app/opt/zbstudio
 
 VERSION=$1
 
 echo Installing ZeroBrane Studio $VERSION...
+mkdir -p /app/bin
 
 if [ -d "$INSTALL_PATH" ]; then
   if [ -e "$INSTALL_PATH/cfg/user.lua" ]; then
     echo "Copying cfg/user.lua file..."
-    cp "$INSTALL_PATH/cfg/user.lua" "app/cfg/user.lua"
+    cp "$INSTALL_PATH/cfg/user.lua" "ZeroBraneStudio-linux/app/cfg/user.lua"
   fi
-  sudo rm -r $INSTALL_PATH
+  rm -r $INSTALL_PATH
 fi
 
 INITIAL_USER=`whoami`
-sudo mkdir -p $INSTALL_PATH
-sudo chown -R $INITIAL_USER $INSTALL_PATH
+mkdir -p $INSTALL_PATH
+mkdir -p /app/bin
+chown -R $INITIAL_USER $INSTALL_PATH
 
-sudo cp -rp app/* $INSTALL_PATH
-
-for ICON_SIZE in 16 22 24 32 48 64 256
-do
-  sudo xdg-icon-resource install --novendor --size $ICON_SIZE $INSTALL_PATH/zbstudio/res/icons/${ICON_SIZE}x${ICON_SIZE}/apps/zbstudio.png "zbstudio"
-done
+cp -rp ZeroBraneStudio-linux/app/* $INSTALL_PATH
 
+cp -rp /app/share/icons/hicolor/ $INSTALL_PATH/zbstudio/res/icons/*
 UPDATE_MENUS="`which update-menus 2>/dev/null`"
 
-sudo xdg-desktop-menu install --novendor $INSTALL_PATH/zbstudio/res/zbstudio.desktop
+cp -p /app/share/applications/ $INSTALL_PATH/zbstudio/res/zbstudio.desktop
 
 if [ -x "$UPDATE_MENUS" ]; then
-  sudo update-menus
+  update-menus
 fi
 
-if [ -e "/usr/bin/zbstudio" ]; then
-  sudo rm /usr/bin/zbstudio
+if [ -e "/app/bin/zbstudio" ]; then
+  rm /app/bin/zbstudio
 fi
-sudo ln -s $INSTALL_PATH/zbstudio.sh /usr/bin/zbstudio
-sudo chmod +x $INSTALL_PATH/zbstudio.sh
-
-echo $VERSION | sudo tee $INSTALL_PATH/VERSION >/dev/null
-
-sudo cp uninstall.sh /usr/bin/zbstudio-uninstall
-sudo chmod +x /usr/bin/zbstudio-uninstall
+ln -s $INSTALL_PATH/zbstudio.sh /app/bin/zbstudio
+chmod +x $INSTALL_PATH/zbstudio.sh
 
-echo Done.
-echo
-echo To uninstall ZeroBrane Studio at a later time, run:
-echo -e '    zbstudio-uninstall'
+echo $VERSION | tee $INSTALL_PATH/VERSION >/dev/null
